### PR TITLE
Created FileSystemWrapper as indicated in issue

### PR DESCRIPTION
**Pull request by zakaluka**

### CODE REVIEW

File: fscd/FileSystemWrapper.fs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            FileSystemEventArgs(WatcherChangeTypes.Deleted, Path.GetDirectoryName re.OldFullPath, Path.GetFileName re.OldFullPath)
+            |> this.OnFileDeleted
+
+    member private this.OnFileCreated(fse:FileSystemEventArgs) =
+        List.iter (fun f -> f fse) this.createHandlers
+
+    member private this.OnFileWatcherError(ee:ErrorEventArgs) =
+        List.iter (fun f -> f ee) this.errorHandlers
+
+    new(path, filter) as this =
```
Comment (dsyme) [inline fscd/FileSystemWrapper.fs]: Can you cahnge this to use an implicit constructor please?

**Comment by dsyme** [comment]
You'll need to add the file to fscd.Tests as well.

Does this actually fix the linked bug?  My impression from the MonoDevelop code is that a separate FSW implementation is needed for OSX

**Comment by nosami** [comment]
> Does this actually fix the linked bug? My impression from the MonoDevelop code is that a separate FSW implementation is needed for OSX

@dsyme: I commented [here](https://github.com/fsprojects/Elmish.XamarinForms/issues/120#issuecomment-405524757)

**Comment by zakaluka** [comment]
Closing as it does not fulfill the goal.

**Pull request closed without merging**